src/python_testing/TC_CHANNEL_5_9.py

```diff
@@ -0,0 +1,150 @@
+        # Priority: NAME > Identifier > AffiliateCallSign > CallSign > Major.Minor
+        if getattr(channel, "name", None):
+            return channel.name
+        if getattr(channel, "identifier", None):
+            return channel.identifier
+        if getattr(channel, "affiliateCallSign", None):
+            return channel.affiliateCallSign
+        if getattr(channel, "callSign", None):
+            return channel.callSign
+        if hasattr(channel, "majorNumber") and hasattr(channel, "minorNumber"):
+            return f"{channel.majorNumber}.{channel.minorNumber}"
+        return None
+
+    @async_test_body
+    async def test_TC_CHANNEL_5_9(self):
+        endpoint = self.user_params.get("endpoint", 1)
+
+        # Step 0: Commissioning already done by test-runner setup
+        self.step("0")
+
+        # Step 1: Read the ChannelList attribute and validate it's a list
+        if pics("CHANNEL.S.A0000"):
+            self.step("1")
+            channel_list = await self.read_channel_list(endpoint)
+            asserts.assert_is_instance(channel_list, list, "ChannelList must be a list.")
+            logging.info(f"Available channels: {channel_list}")
+            # Filter out channels that don't have any matchable identifiers
+            valid_channels = [ch for ch in channel_list if self.extract_channel_identifier(ch)]
+            asserts.assert_true(len(valid_channels) >= 2, "Need at least 2 valid channels for test.")
+        else:
+            self.skip_step("1")
+
+        # Step 2: Read the CurrentChannel attribute to store current state before switching
+        if pics("CHANNEL.S.A0002"):
+            self.step("2")
+            current_channel = await self.read_current_channel(endpoint)
+            logging.info(f"CurrentChannel: {current_channel}")
+            current_identifier = self.extract_channel_identifier(current_channel) if current_channel else None
+        else:
+            self.skip_step("2")
+
+        # Step 3: Send ChangeChannel command with a different valid Match string than current
+        if pics("CHANNEL.S.C00.Rsp"):
+            self.step("3")
+            selected_channel = next(
+                (ch for ch in valid_channels if self.extract_channel_identifier(ch) != current_identifier), None)
+            asserts.assert_is_not_none(selected_channel, "No different channel found for switching.")
+
+            match_str = self.extract_channel_identifier(selected_channel)
+            logging.info(f"Selected match string for ChangeChannel: {match_str}")
+
+            response = await self.send_change_channel_command(endpoint, match_str)
+            asserts.assert_equal(response.status, Status.Success, f"Expected Success status, got {response.status}")
+        else:
+            self.skip_step("3")
+
+        # Step 4: Read CurrentChannel again to verify it has updated to the newly selected channel
+        if pics("CHANNEL.S.A0002"):
```

Comment:
The condition here is depending on a different pics than the step3 ... however logically they seem related: you can only expect a new updated channel if you have sent 'send channel change" in step 3. Why are the PICS codes different? Should this also include the "change channel supported" condition?